Analysis crash with --no-show-progress option.

**Describe the bug**
We are executing an analysis over a folder and we get the next error when use the `--no-show-progress` option:

```
[ERR] Uncaught exception: AggregateException:One or more errors occurred. (One or more errors occurred. (Value cannot be null. (Parameter 'inputStream'))).
at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait(TimeSpan timeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait(TimeSpan timeout)
   at Microsoft.ApplicationInspector.Commands.AnalyzeCommand.DoProcessing(IEnumerable`1 fileEntries, CancellationTokenSource cts) in /mnt/vss/_work/1/s/AppInspector/Commands/AnalyzeCommand.cs:line 1132
   at Microsoft.ApplicationInspector.Commands.AnalyzeCommand.GetResult() in /mnt/vss/_work/1/s/AppInspector/Commands/AnalyzeCommand.cs:line 966
   at Microsoft.ApplicationInspector.CLI.Program.RunAnalyzeCommand(CLIAnalyzeCmdOptions cliOptions) in /mnt/vss/_work/1/s/AppInspector.CLI/Program.cs:line 310
   at Microsoft.ApplicationInspector.CLI.Program.VerifyOutputArgsAndRunAnalyze(CLIAnalyzeCmdOptions options) in /mnt/vss/_work/1/s/AppInspector.CLI/Program.cs:line 124
   at Microsoft.ApplicationInspector.CLI.Program.<>c.<Main>b__1_1(CLIAnalyzeCmdOptions cliAnalyzeCmdOptions) in /mnt/vss/_work/1/s/AppInspector.CLI/Program.cs:line 41
   at CommandLine.ParserResultExtensions.MapResult[T1,T2,T3,T4,T5,TResult](ParserResult`1 result, Func`2 parsedFunc1, Func`2 parsedFunc2, Func`2 parsedFunc3, Func`2 parsedFunc4, Func`2 parsedFunc5, Func`2 notParsedFunc)
   at Microsoft.ApplicationInspector.CLI.Program.Main(String[] args) in /mnt/vss/_work/1/s/AppInspector.CLI/Program.cs:line 40
Error: Process completed with exit code 2.
```

But the thing is that with the same folder and same args without using `--no-show-progress` option it works correctly.
we don't know what file is causing this (is a big java project) With other projects it doesn't happens. 

Thanks in advance.

**To Reproduce**
Steps to reproduce the behavior:
1. Run ApplicationInspector with `-s . -f json --processing-timeout 900000 --file-timeout 300000 -A -u --no-show-progress -r /rules -i -M 20 -o result.json`
2. See error

**Operating Environment (please complete the following information):**
 - Application Inspector Version: 1.9.22
 - OS:  Linux Ubuntu 20.04

--- a/AppInspector.RulesEngine/RuleProcessor.cs
+++ b/AppInspector.RulesEngine/RuleProcessor.cs
@@ -566,11 +566,11 @@ public class RuleProcessor
 
         int startLineNumber =
             start.Line < 0 ? 0 : start.Line > text.LineEnds.Count ? text.LineEnds.Count - 1 : start.Line;
-        int endLineNUmber =
+        int endLineNumber =
             end.Line < 0 ? 0 : end.Line > text.LineEnds.Count ? text.LineEnds.Count - 1 : end.Line;
         // First we try to include the number of lines of context requested
         var excerptStartLine = Math.Max(0, startLineNumber - context);
-        var excerptEndLine = Math.Min(text.LineEnds.Count - 1, endLineNUmber + context);
+        var excerptEndLine = Math.Min(text.LineEnds.Count - 1, endLineNumber + context);
         var startIndex = text.LineStarts[excerptStartLine];
         var endIndex = text.LineEnds[excerptEndLine] + 1;
         // Maximum number of characters to capture on each side

--- a/AppInspector.RulesEngine/TextContainer.cs
+++ b/AppInspector.RulesEngine/TextContainer.cs
@@ -449,12 +449,14 @@ public class TextContainer
 
     /// <summary>
     ///     Returns location (Line, Column) for given index in text
+    ///     If the index is beyond the end of the file, clamps to the end
     /// </summary>
     /// <param name="index"> Position in text (line is one-indexed)</param>
     /// <returns> Location </returns>
     public Location GetLocation(int index)
     {
         for (var i = 1; i < LineEnds.Count; i++)
+        {
             if (LineEnds[i] >= index)
             {
                 return new Location
@@ -463,6 +465,17 @@ public class TextContainer
                     Line = i
                 };
             }
+        }
+
+        // If the index is beyond the end of the file, clamp to the end of the file
+        if (index > LineEnds[^1])
+        {
+            return new Location()
+            {
+                Column = LineEnds[^1] - LineStarts[^1],
+                Line = LineEnds.Count
+            };
+        }
 
         return new Location();
     }

--- a/AppInspector/Commands/AnalyzeCommand.cs
+++ b/AppInspector/Commands/AnalyzeCommand.cs
@@ -4,6 +4,7 @@
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Diagnostics;
 using System.Globalization;
 using System.IO;
@@ -771,21 +772,38 @@ public class AnalyzeCommand
 
                 if (contents != null)
                 {
-                    if (_options.DisableCrawlArchives)
+                    IList<FileEntry> entriesToYield = new List<FileEntry>();
+                    try
                     {
-                        yield return new FileEntry(srcFile, contents);
-                    }
-                    else
-                    {
-                        // Use MemoryStreamCutoff = 1 to force using FileStream with DeleteOnClose for backing, and avoid memory exhaustion.
-                        ExtractorOptions opts = new()
+                        if (_options.DisableCrawlArchives)
                         {
-                            Parallel = false, DenyFilters = _options.FilePathExclusions, MemoryStreamCutoff = 1
-                        };
-                        // This works if the contents contain any kind of file.
-                        // If the file is an archive this gets all the entries it contains.
-                        // If the file is not an archive, the stream is wrapped in a FileEntry container and yielded
-                        foreach (var entry in extractor.Extract(srcFile, contents, opts)) yield return entry;
+                            entriesToYield.Add(new FileEntry(srcFile, contents));
+                        }
+                        else
+                        {
+                            // Use MemoryStreamCutoff = 1 to force using FileStream with DeleteOnClose for backing, and avoid memory exhaustion.
+                            ExtractorOptions opts = new()
+                            {
+                                Parallel = false, DenyFilters = _options.FilePathExclusions, MemoryStreamCutoff = 1
+                            };
+                            // This works if the contents contain any kind of file.
+                            // If the file is an archive this gets all the entries it contains.
+                            // If the file is not an archive, the stream is wrapped in a FileEntry container and yielded
+                            entriesToYield = extractor.Extract(srcFile, contents, opts).ToImmutableList();
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        _logger.LogDebug(
+                            "Failed to analyze file {Path}. {Type}:{Message}. ({StackTrace})",
+                            srcFile, e.GetType(), e.Message, e.StackTrace);
+                        _metaDataHelper?.Metadata.Files.Add(new FileRecord
+                            { FileName = srcFile, Status = ScanState.Error });
+                    }
+
+                    foreach (var entry in entriesToYield)
+                    {
+                        yield return entry;
                     }
                 }
 

